imageprogress/progress_test.go

```diff
@@ -76,12 +70,9 @@ func TestReports(t *testing.T) {
 				test.gen(p)
 				pipeOut.Close()
 			}()
-			var lock sync.Mutex
 			var lastReport report
 			w := newWriter(
 				func(r report) {
-					lock.Lock()
```

Comment:
I'm always leery about removing a lock, but I'm not seeing a reason to keep them.